include/xalloc.h

```diff
@@ -47,6 +48,7 @@ static inline void *xrealloc(void *ptr, const size_t size)
 {
 	void *ret = realloc(ptr, size);
 
+	assert(size);
 	if (!ret && size)
```

Comment:
I'm not sure about it. The attribute __ul_returns_nonnull is probably bad idea here. The ideal solution would be to keep all the xalloc.h to confirm to libc (posix, etc.). It will be difficult to use if the sementic will be different.

Reply:
Commit 54bfeb2117696cee61d82f8bc546bbe906c8f4ea removed abort() and returns nonnull from realloc() because the promise cannot be kept in all situations.